core/src/test/java/org/apache/spark/JavaAPISuite.java

```diff
@@ -63,6 +63,7 @@
 // The test suite itself is Serializable so that anonymous Function implementations can be
 // serialized, as an alternative to converting these anonymous classes to static inner classes;
 // see http://stackoverflow.com/questions/758570/.
+@SuppressWarnings("serial")
```

Comment:
Actually, I put this there to have absolutely zero warnings across the whole project in scala-ide. We should, ihmo, not change the compiler settings because spark uses java serialization by default. Not warning on missing serialization in new classes would broke spark.

But these two classes are tests anyway.

Should we keep it in the PR or remove it?